### PR TITLE
Fix admin permission not revoked when removed from LDAP admin group(Do Not Merge)

### PR DESCRIPTION
**Pull request by stonezdj**
Separate the HasAdminRole(In DB) with the privileges from external auth, and use user.HasAdminPrivilege to check if the current user has admin privilege.
Fix issue: #10012

### CODE REVIEW

File: src/common/dao/role.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func IsAdminRole(userIDOrUsername interface{}) (bool, error) {
 		return false, nil
 	}
 
-	return user.HasAdminRole, nil
+	return user.HasAdminPrivilege(), nil
```
Comment (reasonerjt) [inline src/common/dao/role.go]: dao pkg would better only reflect the data in DB, can we use securityCtx ?

File: src/common/models/user.go [hunk 2]
```diff
@@ -34,15 +34,23 @@ type User struct {
...
-	//	RoleList     []Role `json:"role_list"`
-	HasAdminRole bool      `orm:"column(sysadmin_flag)" json:"has_admin_role"`
-	ResetUUID    string    `orm:"column(reset_uuid)" json:"reset_uuid"`
-	Salt         string    `orm:"column(salt)" json:"-"`
-	CreationTime time.Time `orm:"column(creation_time);auto_now_add" json:"creation_time"`
-	UpdateTime   time.Time `orm:"column(update_time);auto_now" json:"update_time"`
-	GroupIDs     []int     `orm:"-" json:"-"`
-	OIDCUserMeta *OIDCUser `orm:"-" json:"oidc_user_meta,omitempty"`
+	Role         int  `orm:"-" json:"role_id"`
+	HasAdminRole bool `orm:"column(sysadmin_flag)" json:"has_admin_role"`
```
Comment (reasonerjt) [inline src/common/models/user.go]: Can we check how "has_admin_role" in the JSON response is used, I think may be it's better we create another attribute, "sysadmin_flag" to 1:1 map to the DB column and use "HasAdminRole" to reflect to consolidated privilege?

File: src/core/api/user.go [hunk 3]
```diff
@@ -151,7 +151,9 @@ func (ua *UserAPI) Get() {
 		}
 		u.Password = ""
 		if ua.userID == ua.currentUserID {
-			u.HasAdminRole = ua.SecurityCtx.IsSysAdmin()
+			// Because UI is rendered with the condition of HasAdminRole,
+			// copy admin privilege to this field to REST response for UI to render it correctly
+			u.SysAdminFlag = ua.SecurityCtx.IsSysAdmin()
```
Comment (reasonerjt) [inline src/core/api/user.go]: This is wrong, the systemAdminFlag should only reflects the column in DB, but `IsSysAdmin` is the merged result.

File: src/common/dao/dao_test.go [hunk 4]
```diff
@@ -626,6 +626,35 @@ func TestGetRoleByID(t *testing.T) {
 	}
 }
 
+// IsAdminRole returns whether the user is admin.
```
[the remark this answers is not in the document]
Comment (stonezdj) [reply src/common/dao/dao_test.go]: this function retrieves the new state from DB

File: src/common/security/local/context.go [hunk 5]
```diff
@@ -52,13 +52,18 @@ func (s *SecurityContext) GetUsername() string {
 	return s.user.Username
 }
 
+// User get the current user
+func (s *SecurityContext) User() *models.User {
```
Comment (ywk253100) [inline src/common/security/local/context.go]: Is this function used in anywhere?

Comment (ywk253100) [reply src/common/security/local/context.go]: Why we need it?

**Comment by xaleeks** [comment]
what version is this fixed for? can we add a version label on PRs? @stonezdj

**Pull request merged**